crates/merkle-tree/src/contract.rs

```diff
@@ -173,7 +174,7 @@ impl<'tx> StorageCommitmentTree<'tx> {
             block: Some(block),
         };
 
-        let tree = MerkleTree::new(root);
+        let tree = MerkleTree::new(TrieStorageIndex::new(root.get()));
```

Comment:
Do we need the `TrieStorageIndex` conversion here? As far as I've seen `tx.storage_root_index()` is now returning an `Option<TrieStorageIndex>` now so this seems to be redundant.